### PR TITLE
use new_badge_achievement_worker instead of active job

### PR DESCRIPTION
**Pull request by serenaf**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
Next on the list is moving the NewBadgeAchievementJob to a worker. Closely followed a previous PR here, see https://github.com/thepracticaldev/dev.to/pull/5312

I have a problem with the specs, as in my environment the spec file that fails, passes 😅 Might be some flakiness in the system?
## Related Tickets & Documents

https://github.com/thepracticaldev/dev.to/issues/5305

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -62,12 +62,8 @@ def send_new_comment_notifications_without_delay(comment)
     end
 
     def send_new_badge_achievement_notification(badge_achievement)
-      Notifications::NewBadgeAchievementJob.perform_later(badge_achievement.id)
+      Notifications::NewBadgeAchievementWorker.perform_async(badge_achievement.id)
     end
-    # NOTE: this alias is temporary until the transition to ActiveJob is completed
-    # and all old DelayedJob jobs are processed by the queue workers.
-    # It can be removed after pre-existing jobs are done
-    alias send_new_badge_notification send_new_badge_achievement_notification
```
Comment (serenaf) [inline app/models/notification.rb]: Removed this alias, as this job has been in production for a while. Let me know if you want me to put it back in :)

File: spec/models/notification_spec.rb [hunk 2]
```diff
@@ -491,20 +491,12 @@
...
   describe "#send_new_badge_achievement_notification" do
     it "enqueues a new badge achievement job" do
-      assert_enqueued_with(job: Notifications::NewBadgeAchievementJob, args: [badge_achievement.id]) do
+      sidekiq_assert_enqueued_with(job: Notifications::NewBadgeAchievementWorker, args: [badge_achievement.id]) do
         described_class.send_new_badge_achievement_notification(badge_achievement)
       end
     end
   end
 
-  describe "#send_new_badge_notification (deprecated)" do
```
Comment (serenaf) [inline spec/models/notification_spec.rb]: Removed this test as it was part of an intermittent alias, see https://github.com/thepracticaldev/dev.to/pull/5358/files#r363057123

**Comment by serenaf** [comment]
@rhymes thanks so much for the helpful comment 🙇‍♀ I updated the spec in question 🎉 
This should be good to go now 🤞 The two failed specs are fine in my dev environment (i know...works on my machine syndrome 😂 )

**Comment by rhymes** [comment]
@serenaf thanks! Our system tests are a bit on the flay side :(

**Comment by serenaf** [comment]
@mstruve Great catch, thank you! I updated the file with the new commit hook and added an additional test as well :)

**Comment by mstruve** [comment]
Thank you for updating this and adding the spec! However, for this callback we actually want `after_create_commit` because `after_commit` will trigger after every save per [the Rails docs](https://guides.rubyonrails.org/active_record_callbacks.html)
> When a transaction completes, the after_commit or after_rollback callbacks are called for all models created, updated, or destroyed within that transaction. 

and we only want it after the badge is created.

**Comment by serenaf** [comment]
@mstruve Thank you! I did not know this :) Updated the PR one more time :)

**Pull request merged**